java/com/google/flatbuffers/Table.java

```diff
@@ -69,7 +69,8 @@ protected int __offset(int vtable_offset) {
 
   protected static int __offset(int vtable_offset, int offset, ByteBuffer bb) {
     int vtable = bb.array().length - offset;
```

Comment:
I am seeing similar issues in all the new code since I've last looked at this, is there any reason why ``array()`` is accessed directly?  Are you specifically telling the world that only non-sliced on-heap buffers are usable for this feature?

If you are, it should be documented better but I would argue such documentation will mostly be ignored and people will complain on the list about the exception they're getting... if they're lucky to get them.  Off-heap buffers will throw immediately when using the ``array`` method but on-heap slices will most likely only hit at run time.

I am also concerned about the use of ``getShort`` too.  This call will not align correctly so I don't understand why it's being used at all.